src/main/java/com/wavefront/TraceTypePattern.java

```diff
@@ -3,28 +3,66 @@
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 
 import java.util.List;
+import java.util.Random;
 
 @JsonIgnoreProperties(ignoreUnknown = true)
+@SuppressWarnings("unused")
 public class TraceTypePattern {
   public String traceTypeName;
   public int nestingLevel;
-  public int spansNumber;
-  public int tracesNumber;
+  public int tracePercentage;
   public int errorRate;
+  public List<Distribution> spansDistributions;
+//  public LinkedList<Pair<String, LinkedList<String>>> mandatoryTags;
+//  public HashMap<String, LinkedList<String>> optionalTags;
 
-  //    public LinkedList<Pair<String, LinkedList<String> > > mandatoryTags;
-//    public HashMap<String, LinkedList<String>> optionalTags;
-
-  public TraceTypePattern(String traceTypeName, int nestingLevel, int spansNumber,
-                          int tracesNumber, int errorRate) {
+  public TraceTypePattern(String traceTypeName, int nestingLevel, int tracePercentage,
+                          List<Distribution> distributions, int errorRate) {
     this.traceTypeName = traceTypeName;
     this.nestingLevel = nestingLevel;
-    this.spansNumber = spansNumber;
-    this.tracesNumber = tracesNumber;
+    this.tracePercentage = tracePercentage;
+    this.spansDistributions = distributions;
     this.errorRate = errorRate;
   }
 
   public TraceTypePattern() {
 
   }
-}
\ No newline at end of file
+
+  /**
+   * Class represents distribution need for generation of spans and durations per trace type.
+   */
+  public static class Distribution {
+    private final Random random = new Random(System.currentTimeMillis());
```

Comment:
Do we need Random for all instances of Distribution? Maybe it could be "static"?